### PR TITLE
chore: update deno_file to use deno_webidl

### PR DESCRIPTION
**Pull request by lucacasonato**
This changes the custom input converters in deno_file to use deno_webidl
converters.

### CODE REVIEW

**Comment by lucacasonato** [comment]
Unsure why mac CI is failing... builds fine on M1. @bartlomieju could you try to build this locally?

File: op_crates/file/01_file.js [hunk 1]
```diff
@@ -300,42 +344,62 @@
     /**
      * @param {BlobPart[]} fileBits 
      * @param {string} fileName 
-     * @param {FilePropertyBag} [options] 
+     * @param {FilePropertyBag} options
```
Comment (kitsonk) [inline op_crates/file/01_file.js:347]:
```suggestion
     * @param {FilePropertyBag=} options 
```

**Review by kitsonk — COMMENTED** [review]
I am getting build failures locally on my Intel mac.  Not sure why.

**Comment by bartlomieju** [comment]
I'm getting a segfault when building on mac:
```
signal: 11, SIGSEGV: invalid memory reference
```

**Comment by lucacasonato** [comment]
Kitson was getting that too. No idea where it is coming from. I can't reproduce on M1 or Linux. Is it segfaulting during snapshotting?

**Comment by bartlomieju** [comment]
> Kitson was getting that too. No idea where it is coming from. I can't reproduce on M1 or Linux. Is it segfaulting during snapshotting?

I think it is coming from snapshot:
```
   Compiling deno_runtime v0.10.1 (/Users/biwanczuk/dev/deno/runtime)
       Fresh swc_ecmascript v0.24.1
       Fresh swc_bundler v0.25.1
     Running `/Users/biwanczuk/dev/deno/target/debug/build/deno_runtime-5f47a5fb67edbe5c/build-script-build`
       Fresh dprint-swc-ecma-ast-view v0.10.0
       Fresh deno_doc v0.1.23
       Fresh deno_lint v0.2.19
       Fresh dprint-plugin-typescript v0.41.0
[deno_runtime 0.10.1] cargo:rustc-env=TARGET=x86_64-apple-darwin
[deno_runtime 0.10.1] cargo:rustc-env=PROFILE=debug
error: failed to run custom build command for `deno_runtime v0.10.1 (/Users/biwanczuk/dev/deno/runtime)`

Caused by:
  process didn't exit successfully: `/Users/biwanczuk/dev/deno/target/debug/build/deno_runtime-5f47a5fb67edbe5c/build-script-build` (signal: 11, SIGSEGV: invalid memory reference)
  --- stdout
  cargo:rustc-env=TARGET=x86_64-apple-darwin
  cargo:rustc-env=PROFILE=debug
```

**Comment by kitsonk** [comment]
A bit of experimenting and it is this line: https://github.com/denoland/deno/pull/10042/files#diff-4e14bca696f63fff1ff474012a00ccefd7f92183d547304e97c3fc2af5c2cfdeR624

It is touching `.locale` during snapshotting, which is touching ICU, which we believe has duplicate symbols because of the WebGPU stuff, which on non-M1 macs we know is a problem on debug builds.

File: op_crates/file/01_file.js (hunk 1 above)
Comment (lucacasonato) [reply op_crates/file/01_file.js:347]: Results in `options` is possibly undefined, which is wrong.

**Review by kitsonk — APPROVED** [review]
LGTM

**Pull request merged**